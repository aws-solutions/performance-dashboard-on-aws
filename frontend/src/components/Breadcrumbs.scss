@import "../styles/base";

.usa-breadcrumb {
    background-color: transparent !important;
    @include u-padding-bottom($theme-breadcrumb-padding-bottom);
    @include u-padding-top($theme-breadcrumb-padding-top);
}

// Breadcrumb wrap
.usa-breadcrumb--wrap {
    @include u-line-height($theme-breadcrumb-font-family, 4);
}
.usa-breadcrumb__list-item {
    @include u-white-space("wrap");

    // If parent link only, only show parent of current link
    &:nth-last-child(2) {
        @include not-sr-only;

        .usa-breadcrumb__link {
            @include button-unstyled;
            @include exdent-icon;
            @include place-icon(
                $icon-breadcrumb-back,
                "before",
                0,
                baseline,
                $theme-breadcrumb-background-color
            );

            // Override link colors from button-unstyled()
            @include set-link-from-bg(
                $theme-breadcrumb-background-color,
                $theme-breadcrumb-link-color,
                $context: $breadcrumb-context
            );

            @include u-display("inline-block");
            @include u-padding-bottom($theme-breadcrumb-padding-bottom);
            @include u-padding-top($theme-breadcrumb-padding-top);

            &:before {
                bottom: $icon-vertical-spacer;
                // Magic number to center icon
                height: $breadcrumb-icon-display-height;
                position: relative;
            }

            // Prevent underline that extends beyond text
            &,
            &:hover,
            &:active {
                @include u-text("no-underline");
            }
            span {
                @include u-text("underline");
            }
        }

        // Override icon spacing from place-icon() with non-token value
        .usa-breadcrumb__link::before {
            margin-right: $breadcrumb-icon-spacing;
        }
    }
    &:not(:last-child)::after {
        display: none !important;
    }
    &:not(:nth-last-child(2)) {
        display: none !important;
    }
}
