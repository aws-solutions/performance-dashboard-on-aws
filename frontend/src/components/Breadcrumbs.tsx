/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { Link } from "react-router-dom";
import "./Breadcrumbs.scss";
import { useTranslation } from "react-i18next";

interface Props {
    crumbs: Array<{
        label?: string;
        url?: string;
    }>;
}

function Breadcrumbs(props: Props) {
    const { t } = useTranslation();
    return (
        <div className="padding-1">
            <nav className="usa-breadcrumb usa-breadcrumb--wrap" aria-label={t("Breadcrumbs")}>
                <ol className="usa-breadcrumb__list">
                    {props.crumbs?.map((crumb, index) => {
                        return crumb.url ? (
                            <li className="usa-breadcrumb__list-item" key={crumb.label || index}>
                                <Link to={crumb.url} className="usa-breadcrumb__link">
                                    <span>{crumb.label}</span>
                                </Link>
                            </li>
                        ) : (
                            <li
                                className="usa-breadcrumb__list-item usa-current"
                                aria-current="page"
                                key={crumb.label || index}
                            >
                                <span>{crumb.label}</span>
                            </li>
                        );
                    })}
                </ol>
            </nav>
        </div>
    );
}

export default Breadcrumbs;
