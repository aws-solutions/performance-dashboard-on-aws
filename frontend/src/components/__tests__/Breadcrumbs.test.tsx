/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Breadcrumbs from "../Breadcrumbs";

test("renders a breadcrumb with no crumbs", async () => {
    const wrapper = render(<Breadcrumbs crumbs={[]} />, {
        wrapper: MemoryRouter,
    });
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a breadcrumb with one crumb", async () => {
    const wrapper = render(
        <Breadcrumbs
            crumbs={[
                {
                    label: "Add content",
                },
            ]}
        />,
        { wrapper: MemoryRouter },
    );
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a breadcrumb with two crumbs", async () => {
    const wrapper = render(
        <Breadcrumbs
            crumbs={[
                {
                    label: "Home",
                    url: "/home",
                },
                {
                    label: "Add content",
                },
            ]}
        />,
        { wrapper: MemoryRouter },
    );
    expect(wrapper.container).toMatchSnapshot();
});
