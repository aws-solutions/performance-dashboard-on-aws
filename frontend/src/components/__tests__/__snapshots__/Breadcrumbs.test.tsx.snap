// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a breadcrumb with no crumbs 1`] = `
<div>
  <div
    class="padding-1"
  >
    <nav
      aria-label="Breadcrumbs"
      class="usa-breadcrumb usa-breadcrumb--wrap"
    >
      <ol
        class="usa-breadcrumb__list"
      />
    </nav>
  </div>
</div>
`;

exports[`renders a breadcrumb with one crumb 1`] = `
<div>
  <div
    class="padding-1"
  >
    <nav
      aria-label="Breadcrumbs"
      class="usa-breadcrumb usa-breadcrumb--wrap"
    >
      <ol
        class="usa-breadcrumb__list"
      >
        <li
          aria-current="page"
          class="usa-breadcrumb__list-item usa-current"
        >
          <span>
            Add content
          </span>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;

exports[`renders a breadcrumb with two crumbs 1`] = `
<div>
  <div
    class="padding-1"
  >
    <nav
      aria-label="Breadcrumbs"
      class="usa-breadcrumb usa-breadcrumb--wrap"
    >
      <ol
        class="usa-breadcrumb__list"
      >
        <li
          class="usa-breadcrumb__list-item"
        >
          <a
            class="usa-breadcrumb__link"
            href="/home"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          aria-current="page"
          class="usa-breadcrumb__list-item usa-current"
        >
          <span>
            Add content
          </span>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;
